actions/nightly-build-publish/main.sh

```diff
@@ -0,0 +1,155 @@
+#!/bin/bash
+
+if [ -z "$VERSION" ]; then
+  echo "You must set the VERSION environment variable" >&2
+  exit 1
+fi
+
+if [ -z "$NPM_TOKEN" ]; then
+  echo "You must set the NPM_TOKEN environment variable" >&2
+  exit 1
+fi
+
+GITHUB_TOKEN=${GITHUB_TOKEN:-}
+LLAMA_STACK_ONLY=${LLAMA_STACK_ONLY:-false}
+
+source $(dirname $0)/../common.sh
+
+set -euo pipefail
+set -x
+
+is_truthy() {
+  case "$1" in
+  true | 1) return 0 ;;
+  false | 0) return 1 ;;
+  *) return 1 ;;
+  esac
+}
+
+setup_environment() {
+  echo "Setting up build environment..."
+
+  npm config set '//registry.npmjs.org/:_authToken' "$NPM_TOKEN"
+
+  npm install -g yarn
+
+  TMPDIR=$(mktemp -d)
+  cd $TMPDIR
+
+  uv venv -p python3.12
+  source .venv/bin/activate
+  uv pip install twine
+
+  install_dependencies  # Installs test dependencies
+}
+
+clone_and_prepare_repo() {
```

Comment:
maybe I'm missing something, why is this method needed when actions/checkout@v4 exists?